sgkit/api.py

```diff
@@ -0,0 +1,70 @@
+from typing import Any, Hashable, Mapping
+
+import xarray as xr
+
+from .utils import check_array_like
+
+DIM_VARIANT = "variant"
+DIM_SAMPLE = "sample"
+DIM_PLOIDY = "ploidy"
+DIM_ALLELE = "allele"
+DIM_GENOTYPE = "genotype"
+
+
+def create_genotype_call_dataset(
+    variant_contig: Any,
+    variant_pos: Any,
+    variant_alleles: Any,
+    sample_id: Any,
+    call_gt: Any,
+    call_gt_phased: Any = None,
+    variant_id: Any = None,
+) -> xr.Dataset:
+    """Create a dataset of genotype calls.
+
+    Parameters
+    ----------
+    variant_contig : array_like, np.int8
+        The (index of the) contig for each variant.
+    variant_pos : array_like, np.int32
+        The reference position of the variant.
+    variant_alleles : array_like, S1
+        The possible alleles for the variant.
+    sample_id : array_like, str
+        The unique identifier of the sample.
+    call_gt : array_like, np.int8
+        Genotype, encoded as allele values (0 for the reference, 1 for
+        the first allele, 2 for the second allele), or -1 to indicate a
+        missing value.
+    call_gt_phased : array_like, bool, optional
+        A flag for each call indicating if it is phased or not. If
+        omitted all calls are unphased.
+    variant_id: array_like, str, optional
+        The unique identifier of the variant.
+
+    Returns
+    -------
+    xr.Dataset
+        The dataset of genotype calls.
+
+    """
+    check_array_like(variant_contig, kind="i", ndim=1)
```

Comment:
I know I mentioned 8-bit ints for this since the huge majority of datasets don't actually make use of all the alt contigs that there are now in newer reference builds but there are definitely some situations where 128 contigs wouldn't be enough.  We could just ignore that for now but I thought I'd point out that it may make sense for this function to do things like take any data type for positions, contig indexes, and calls and carefully coerce them to something more parsimonious as a convenience.

Reply:
The checks are actually quite loose in that they just check the dtype's kind (integer, string, boolean, etc), not its width (8 bit, 32 bit, etc). I should change the documentation to reflect that though since it currently says `np.int8`.